### ISSUE TITLE
Provide a shouldDelete hook in Qualifier interface

### ISSUE THREAD

**Issue by jian-he** [kubernetes-incubator/external-storage#1044]
Currently, the Qualifier has a ShouldProvision hook for external-provisioner to determine whether shouldProvision or not.
```
// Qualifier is an optional interface implemented by provisioners to determine
// whether a claim should be provisioned as early as possible (e.g. prior to
// leader election).
type Qualifier interface {
	// ShouldProvision returns whether provisioning for the claim should
	// be attempted.
	ShouldProvision(*v1.PersistentVolumeClaim) bool
}
```
Similarly, we can add a shouldDelete hook for external-provisioner to determine whether to delete the PV or not.  - we implemented a custom provisioner and would like to inject  logic for deletion such as time-based or space-based deletion policy.
```
shouldDelete(volume *v1.PersistentVolume) bool
```

@jsafrane ,  does this make sense ?

**Comment by wongma7** [comment]
yes, can be added. note that in the meantime there is also the option for Delete to return IgnoredError. The difference is that ShouldProvision/ShouldDelete gets called before Delete even gets called.

**Comment by zhangxiaoyu-zidif** [comment]
great！ we can add this method to avoid many outer processes. thumbs up.
